### PR TITLE
housekeeping: Update loggers testing and make NLog a full logger

### PR DESCRIPTION
**Pull request by glennawatson**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
housekeeping

**What is the current behavior? (You can also link to an open issue here)**
That the splat NLog adapter uses the ILogger interface then use the WrappingLogger. This doesn't allow us to take advantage of structured logging introduced in NLog v4.5

https://github.com/NLog/NLog/wiki/How-to-use-structured-logging

**What is the new behavior (if this is a feature change)?**
Now use the IFullLogger interface. This allows the full types to be passed to NLog instead of us processing with string.Format's, and allows like the web service target in NLog transform into JSON.

**What might this PR break?**
Shouldn't break anything.

**Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/Splat.NLog/NLogLogger.cs [hunk 1]
```diff
@@ -98,6 +113,666 @@ public void Write(Exception exception, string message, Type type, LogLevel logLe
...
+        /// <inheritdoc/>
+        public void Debug(string message)
+        {
+            _inner.Debug(message);
+        }
+
+        /// <inheritdoc/>
+        public void Debug<T>(string message)
+        {
+            _inner.Factory.GetLogger(typeof(T).ToString()).Debug(message);
```
Comment (dpvreony) [inline src/Splat.NLog/NLogLogger.cs]: Would it be worthwhile using an MRU cache?

Comment (dpvreony) [inline src/Splat.NLog/NLogLogger.cs]: Would it be worthwhile using an MRU cache? for _inner.Factory.GetLogger(…)
i'll just mark the one of them. so don't need to click resolve 15 times :)

Comment (glennawatson) [reply src/Splat.NLog/NLogLogger.cs]: Probably not until https://github.com/reactiveui/splat/issues/266 is fixed which is next on my to-do list. Along with fixing the tests for the full loggers which I will do tonight

Comment (glennawatson) [reply src/Splat.NLog/NLogLogger.cs]: I will address #266 and come back to this.

Comment (dpvreony) [reply src/Splat.NLog/NLogLogger.cs]: I've just raised a PR for 266

**Pull request merged**